app/assets/javascripts/discourse/models/user.js.es6

```diff
@@ -367,20 +367,20 @@ const User = RestModel.extend({
     });
   },
 
-  toggleSecondFactor(token, enable, method) {
+  toggleSecondFactor(authToken, authMethod, targetMethod, enable) {
     return ajax("/u/second_factor.json", {
       data: {
-        second_factor_token: token,
-        second_factor_method: method,
+        second_factor_auth: [authToken, authMethod],
```

Comment:
I think that changing this to an array makes it confusing in terms of what values the API expects. Is there a reason we need to make it an array instead of using one key per value?

Reply:
I used an array to group together 2fa method and key we're authenticating with and 2fa that should be modified. I will split the array in separate params then to avoid confusion.